### ISSUE TITLE
(foo##bar).baz prints wrongly

### ISSUE THREAD

**Issue by chenglou** [facebook/reason#774]
Prints as `foo##bar.baz`

Error: `Js object ## expect syntax like obj##(paint (a,b))`

@bsansouci @Schmavery

**Comment by jordwalke** [comment]
Hmmm the printer thinks that dot is lower precedence than ## but the parser knows that dot is highe precedence than ## it seems.

**Comment by thangngoc89** [comment]
Here is a slightly different version with the same error: 

```
let foo = {|{"Uppercase": "foo"}|};

let parsed = foo |> Js.Json.parseExn;

Js.log(parsed##Uppercase);
```

**Comment by anmonteiro** [comment]
I think this was fixed in #2050
